process/transaction/baseProcess.go

```diff
@@ -150,6 +150,7 @@ func (txProc *baseTxProcessor) checkTxValues(
 		txFee = txProc.economicsFee.ComputeTxFee(tx)
 	}
 
+	// TODO: check this: if checks the txFee against the balance before the txFee is finalized on the next if
```

Comment:
this should stay is it is for backward compatibility reasons for the period until flagPenalizedTooMuchGas is not sent

Reply:
ok! removed